### PR TITLE
Census Response Tests Should Not Rely on the Real Census API (Resolves #55)

### LINKED ISSUE #55: Census Response Tests Should Not Rely on the Real Census API
These tests can take around 30 seconds to run which can really slow down iteration time.

We should be able to replace usage of the real API with test fixtures like:

```
[["NAME","B03002_001E","B03002_005E","state","zip code tabulation area"],
["ZCTA5 72827","166","0","05","72827"],
["ZCTA5 72834","10642","53","05","72834"],
["ZCTA5 72845","1018","0","05","72845"],
["ZCTA5 72860","611","0","05","72860"],
["ZCTA5 72901","20985","271","05","72901"],
```

and 

```
[["NAME","B03002_001E","B03002_005E","state","county"],
["Sangamon County, Illinois","197661","168","17","167"],
["Randolph County, Illinois","32546","47","17","157"],
["Ogle County, Illinois","51328","145","17","141"],
["Morgan County, Illinois","34426","63","17","137"],
["White County, Illinois","14025","5","17","193"],
```

We'll have to make sure the test fixtures cover all the edge cases in the data. e.g. there's some null values in the poverty dump response and we'll have to make sure we replicate that case.

### PR DESCRIPTION
**Pull request by StephenWoo**
Uses snapshot JSONs of the current census API responses rather than querying the API directly.

### CODE REVIEW

File: src/census_response.py [hunk 1]
```diff
@@ -254,11 +264,14 @@ def get_census_data(request: CensusRequest, geography_type: str) -> data.Wrapper
     return wrapper
 
 
-def get_census_data_list(data_requests: List[CensusRequest], geo_ls: List[str] = ["zip", "county"]) -> data.Wrapper:  # noqa: E501
+def get_census_data_list(data_requests: List[CensusRequest], geo_ls: List[str] = ["zip", "county"], mock_responses: Dict[str, str] = None) -> data.Wrapper:  # noqa: E501
```
Comment (brickman1444) [inline src/census_response.py:267]: Same here, if you get mypy errors, try Optional[Dict[str, str]]

File: tests/resources/test_poverty_dump_input_county.json [hunk 2]
```diff
@@ -0,0 +1,103 @@
+[["NAME","S1701_C01_001E","S1701_C02_001E","S1701_C02_002E","state","county"],
```
Comment (brickman1444) [inline tests/resources/test_poverty_dump_input_county.json:1]: I would take out "dump" from the file name of these four test data files. The "dump" versus "merge" is just for the output format and it looks like you're reusing the same input for both the merge and dump tests so there's no need to differentiate.

**Review by brickman1444 — APPROVED** [review]
Looks good. Thanks for taking care of this!

Do the census tests run a lot faster now?

Merge in main and you won't get that binning error any more.

After this change the pretty-printed output change are both in, I think we should make another change to reduce the size of the input for the tests. All our other tests use much smaller input data which helps keep the files small, the tests fast and the tests easy to manually verify.

File: src/census_response.py [hunk 3]
```diff
@@ -45,7 +45,8 @@ def download_census_data(geo_ls=["zip", "county"]) -> data.Wrapper:
 
 def get_census_response(table_url: str,
                         get_ls: Collection[str],
-                        geo: str) -> List[List[str]]:
+                        geo: str,
+                        mock_response: str = None) -> List[List[str]]:
```
[the remark this answers is not in the document]
Comment (StephenWoo) [reply src/census_response.py:49]: No mypy error, but I do prefer the `Optional` notation

File: tests/resources/test_poverty_dump_input_county.json (hunk 2 above)
Comment (StephenWoo) [reply tests/resources/test_poverty_dump_input_county.json:1]: Will do!

**Comment by StephenWoo** [comment]
Tests run much faster now!
